src/CarbonAware.WebApi/test/unitTests/CarbonAwareControllerTests.cs

```diff
@@ -120,5 +121,18 @@ public async Task GetBestEmissions_EmptyResultReturnsNoContent()
         TestHelpers.AssertStatusCode(ar, HttpStatusCode.NoContent);
     }
 
+    /// <summary>
+    /// Tests empty or null location arrays throw ArgumentException.
+    /// </summary>
+    [TestCase(new object?[] { null, null }, TestName = "array of nulls: simulates 'location=&location=' empty value input")]
+    [TestCase(new object?[] { null, }, TestName = "array of nulls: simulates 'location=' empty value input")]
+    [TestCase(new object?[] { }, TestName = "empty array: simulates no 'location' query string")]
+    public void GetEmissions_NoLocations_ThrowsException(params string[] locations)
```

Comment:
It would be good to have another case where `new object[] { "fake location", null }` to be catch.

Reply:
We can add that case, but it wont belong in this block, since that is a valid request where we do get "fake location" and ignore the null.  Should return 200 (if the location exists)